OOP/assignment3/RailwayReservationSystem.java

```diff
@@ -0,0 +1,321 @@
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Scanner;
+
+public class RailwayReservationSystem {
+	public static ArrayList<Train> trainList = new ArrayList<Train>();
+
+	private static void prepareTrain() {
+
+		final String COMMA_DELIMITER = ",";
+		final String url = "D://Train/TrainDetails.csv";
+
+		BufferedReader fileReader = null;
+		try {
+
+			String line = "";
+			fileReader = new BufferedReader(new FileReader(url));// file reading
+			fileReader.readLine();
+
+			while ((line = fileReader.readLine()) != null) {
+
+				Train train = new Train();
+				String[] tokens = line.split(COMMA_DELIMITER);// separator
+
+				if (tokens.length > 0) {
+					train.setName(tokens[0]);
+					train.setFromto(tokens[1]);
+					train.setDuration(Integer.parseInt(tokens[2]));
+					train.setAvail(Integer.parseInt(tokens[3]));
+
+					if (tokens[4].equals("PASSENGER"))
```

Comment:
Although I missed this in preview review, instead of : - 
   tokens[4].equals("PASSENGER")
we should better use : -
   tokens[4].equals(Train.Type.PASSENGER.toString())
In the latter case JVMs string literal pool is saved, and it is meaningful too. What do you say? :)